MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1357,114 @@ extension NavigationMapView {
     }
 }
 
+// MARK: - Building Extrusion Highlights
+
+extension NavigationMapView {
+    public func highlightBuildingExtrusion(for coordinate: CLLocationCoordinate2D) {
+        if let buildingId = buildingId(coordinate: coordinate) {
+            extrudeBuildings([(buildingId, buildingExtrusionColor)], extrudeAll: true)
+        } else {
+            unhighlightBuildingExtrusions()
+        }
+    }
+    
+    public func unhighlightBuildingExtrusions() {
+        extrudeBuildings([(0, buildingExtrusionColor)], extrudeAll: true)
+    }
+    
+    private func addStyleLayersIfNecessary() {
+        if let buildingsSource = style?.source(withIdentifier: "composite") {
+            
+            if let highlightedBuildingLayer = highlightedBuildingLayer {
+                style?.removeLayer(highlightedBuildingLayer)
+            }
+            
+            highlightedBuildingLayer = MGLFillExtrusionStyleLayer(identifier: StyleLayerIdentifier.buildingExtrusion, source: buildingsSource)
+            highlightedBuildingLayer?.sourceLayerIdentifier = "building"
+            highlightedBuildingLayer?.fillExtrusionColor = NSExpression(forConstantValue: UIColor.white)
+            highlightedBuildingLayer?.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+            highlightedBuildingLayer?.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+  
+            if let highlightedBuildingLayer = highlightedBuildingLayer {
+                style?.addLayer(highlightedBuildingLayer)
+            }
+        }
+    }
+
+    private func buildingId(coordinate: CLLocationCoordinate2D) -> Int? {
+        let screenCoordinate = convert(coordinate, toPointTo: self)
+        let features = visibleFeatures(at: screenCoordinate)
+        
+        if let feature = features.first, let buildingId = feature.identifier as? Int {
+            return buildingId
+        }
+        
+        return nil
+    }
+    
+    private func extrudeBuildings(_ buildings: [(Int, UIColor)], extrudeAll: Bool) {
```

Comment:
I don’t think we can assume that the building ID is integral or that it fits into 32&nbsp;bits on 32-bit architectures. The predicates in this method would work regardless of the feature identifier’s type; strings would work too, as long as we pass the identifiers into `NSPredicate(format:arguments:)` instead of using string interpolation. But even if we were to assume that the identifiers are integral, it would be possible for them to overflow the `Int` type. [`UInt64` or `Int64`](https://docs.mapbox.com/ios/api/maps/6.0.0/working-with-geojson-data.html) would be a safer representation.